core/state/reader.go

```diff
@@ -48,6 +48,19 @@ type Reader interface {
 	// - The returned storage slot is safe to modify after the call
 	Storage(addr common.Address, slot common.Hash) (common.Hash, error)
 
+	// ContractCode returns the code associated with a particular account.
+	//
+	// - Returns an empty code if it does not exist
+	// - It can return an error to indicate code doesn't exist
+	// - The returned code is safe to modify after the call
+	ContractCode(addr common.Address, codeHash common.Hash) ([]byte, error)
+
+	// ContractCodeSize returns the size of the code associated with a particular account.
+	//
+	// - Returns 0 if the code does not exist
+	// - It can return an error to indicate code doesn't exist
```

Comment:
What do you mean "can". Either it does or it does not. E.g " It returns an error if the account does not exist in the state".